HW0/P4/P4.py

```diff
@@ -40,37 +45,93 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_obs = np.loadtxt('P4_measurements.txt', delimiter=',')
+    x_coords = [row[0]/rx for row in s_obs]
+    y_coords = [row[1]/ry for row in s_obs]
+    z_coords = [row[2]/rz for row in s_obs]
+    ax.plot(x_coords, y_coords, z_coords,'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1, 0, 0, dt, 0, 0],
+                   [0, 1, 0, 0, dt, 0],
+                   [0, 0, 1, 0, 0, dt],
+                   [0, 0, 0, 1-c*dt, 0, 0],
+                   [0, 0, 0, 0, 1-c*dt, 0],
+                   [0, 0, 0, 0, 0, 1-c*dt]])
```

Comment:
Are you familiar with the np.diag() command? I've found that it makes initializing "sparse" matrices really easy and less tedious. Check it out!